src/DropAutomation.sol

```diff
@@ -0,0 +1,288 @@
+
+    /// @notice Helper mapping to check if a token is part of the swap set
+    mapping(address => bool) public isSwapToken;
+
+    /// @notice Tick spacing for each swap token's pool with MAMO
+    mapping(address => int24) public swapTickSpacing;
+
+    /// @notice Maximum slippage tolerance in basis points applied to each swap
+    uint256 public maxSlippageBps;
+
+    uint256 private constant BPS_DENOMINATOR = 10_000;
+    uint256 private constant MAX_SLIPPAGE_CAP_BPS = 500; // 5% maximum allowed slippage
+    uint256 private constant DEFAULT_SLIPPAGE_BPS = 100; // 1% default slippage
+    uint256 private constant SWAP_DEADLINE_BUFFER = 300; // 5 minutes deadline buffer for MEV protection
+    int24 private constant MAMO_CBBTC_TICK_SPACING = 200; // Tick spacing for MAMO/cbBTC CL pool
+
+    event DropCreated(uint256 mamoAmount, uint256 cbBtcAmount);
+    event DedicatedMsgSenderUpdated(address indexed oldSender, address indexed newSender);
+    event SwapTokenAdded(address indexed token, int24 tickSpacing);
+    event SwapTokenRemoved(address indexed token);
+    event SwapRouteUpdated(address indexed token, int24 tickSpacing);
+    event TokensSwapped(address indexed token, uint256 amountIn, uint256 amountOut);
+    event MaxSlippageUpdated(uint256 oldValueBps, uint256 newValueBps);
+
+    error NotDedicatedSender();
+    error InvalidSlippage();
+
+    modifier onlyDedicatedMsgSender() {
+        if (msg.sender != dedicatedMsgSender) {
+            revert NotDedicatedSender();
+        }
+        _;
+    }
+
+    constructor(
+        address owner_,
+        address dedicatedMsgSender_,
+        address mamoToken_,
+        address cbBtcToken_,
+        address fMamoSafe_,
+        address safeRewardsDistributorModule_
+    ) Ownable(owner_) {
+        require(owner_ != address(0), "Invalid owner");
+        require(dedicatedMsgSender_ != address(0), "Invalid dedicated sender");
+        require(mamoToken_ != address(0), "Invalid MAMO token");
+        require(cbBtcToken_ != address(0), "Invalid cbBTC token");
+        require(fMamoSafe_ != address(0), "Invalid F-MAMO safe");
+        require(safeRewardsDistributorModule_ != address(0), "Invalid rewards module");
+
+        dedicatedMsgSender = dedicatedMsgSender_;
+        MAMO_TOKEN = IERC20(mamoToken_);
+        CBBTC_TOKEN = IERC20(cbBtcToken_);
+        F_MAMO_SAFE = fMamoSafe_;
+        SAFE_REWARDS_DISTRIBUTOR_MODULE = IRewardsDistributorSafeModule(safeRewardsDistributorModule_);
+        maxSlippageBps = DEFAULT_SLIPPAGE_BPS;
+    }
+
+    /**
+     * @notice Returns the list of tokens configured for MAMO swaps
+     */
+    function getSwapTokens() external view returns (address[] memory) {
+        return swapTokens;
+    }
+
+    /**
+     * @notice Main automation entry point
+     * @dev Executes earn cycle, swaps configured tokens to MAMO, and forwards rewards to the Safe
+     */
+    function createDrop() external onlyDedicatedMsgSender {
```

Comment:
I think this function should also call the burn and earn contracts, split, etc. And this contract must also be compatible with Aerodrome staked LPs and then call `getReward`